### DISCUSSION TITLE
Support YouTube feed 📺

### DISCUSSION THREAD

**Discussion by chuanqisun** [osmoscraft/osmosfeed#47] [Ideas]
It turns out YouTube comes with Atom feed at `https://www.youtube.com/feeds/videos.xml?channel_id=<channel_id>`
There is [a good web component](https://github.com/paulirish/lite-youtube-embed) that can render a YouTube video given its ID, which can be extracted from the url in each feed "article".

Combining the two, we should be able to render a YouTube feed without any code change. I plan to build an example soon-ish.

**Comment by ghost** [comment]
Yes @chuanqisun kindly add youtube support

**Comment by chuanqisun** [comment]
I'm working on this right now. The plan is to have a template dedicated to displaying YouTube content. I have to keep the scope focused to YouTube without mixing other types of content because the UX for YouTube channel browsing differs significantly from textual content browsing. Mixed content type will be supported in a future effort and leverage the component and learning from this effort.

Progress so far:
- [x] Organize videos by YouTube channel, sort by most recent updates
- [x] Validate YouTube embed compatibility
- [x] Technical design for mobile + desktop horizontal scroll and prev/next button
- [x] Styling


A sneak peek of the prototype:

![image](https://user-images.githubusercontent.com/1895289/122699964-4508c380-d1ff-11eb-819e-0f32c89b6535.png)

**Reply by ghost** [reply]
Looks nice! Thank you very much for your efforts. Count me in for any beta testing if needed. Waiting eagerly for release.

**Reply by chuanqisun** [reply]
I got a working version. [Demo](https://osmoscraft.github.io/osmosfeed-examples/youtube-dark/) | [Source](https://github.com/osmoscraft/osmosfeed-examples/tree/main/examples/youtube-dark)

You can copy/paste the `includes` and `static` directory from the [example](https://github.com/osmoscraft/osmosfeed-examples/tree/main/examples/youtube-dark) to get started. Since the work is done on the templating layer, it's not as easy as plug & play. So apologies for the inconvenience.

Also, the template only handles YouTube sources. If you mix in other sources, the layout will break. Something I hope to improve in future iterations by adding built-in content type detection to the engine.

Let me know how this works or doesn't work for you. I'm eager to hear your feedback. 

![image](https://user-images.githubusercontent.com/1895289/123044182-2e549f00-d3ae-11eb-9efd-0ad8b7021c66.png)

**Reply by ghost** [reply]
Thanks @chuanqisun ! No need for apologies. I created a new repo from osmosfeed and added those folders. For anyone referring this, you also need to add top 2-3 lines in osmosfeed.yaml from example.

Its running very smooth on mobile and pc. Sorting is intelligent. Eagerly waiting to get both layouts merged.
